### PR TITLE
📖 release: restructure release docs team roles

### LINKED ISSUE #10354: Restructure release docs stack
here is the kubernetes release example: https://github.com/kubernetes/sig-release/tree/master/release-team

the idea is to split some of these lengthier docs (i.e. https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/release/release-tasks.md) so they are broken up by team.  this makes them easier to consume and improves the organization of the release docs

/area release

### PR DESCRIPTION
**Pull request by dhij**
**What this PR does / why we need it**:

This PR breaks up the lengthy [release tasks doc](https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/release/release-tasks.md) by team as suggested in https://github.com/kubernetes-sigs/cluster-api/issues/10354

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #10354

### CODE REVIEW

File: docs/release/role-handbooks/ci-signal/README.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+## Overview
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Responsibilities](#responsibilities)
+- [Tasks](#tasks)
+  - [Setup jobs and dashboards for a new release branch](#setup-jobs-and-dashboards-for-a-new-release-branch)
+  - [[Continuously] Monitor CI signal](#continuously-monitor-ci-signal)
+    - [[Continuously] Reduce the amount of flaky tests](#continuously-reduce-the-amount-of-flaky-tests)
```
Comment (chrischdi) [inline docs/release/role-handbooks/ci-signal/README.md:12]:
```suggestion
  - [[Continuously] Reduce the amount of flaky tests](#continuously-reduce-the-amount-of-flaky-tests)
```

**Comment by chrischdi** [comment]
> @chrischdi I fixed the broken links and would appreciate if you could have a look 🙏
> 
> When this PR gets merged, I will need a follow-up PR to fix a bunch of broken links to `release-tasks.md` since that file would no longer exist (ex. in `.github/ISSUE_TEMPLATE`). I saw that including those link changes in this PR will break the CI since those are absolute `https://` links.

Makes sense, maybe you can already prepare that PR so we can merge it quickly after this one to not break main for too long.

File: docs/release/release-tasks.md [hunk 2]
```diff
@@ -1,547 +0,0 @@
...
-   <br>Prior art: [Tasks for v1.6 release cycle](https://github.com/kubernetes-sigs/cluster-api/issues/9094)
-2. Create a release improvement tasks [GitHub Project Board](https://github.com/orgs/kubernetes-sigs/projects/55) to track 
-   the current status of all improvement tasks planned for the release, their priorities, status (i.e `Done`/`In Progress`)
-   and to distribute the work among the Release Team members.
-
-   **Notes**:
-   * At the beginning of the cycle, Release Team Lead should prepare the improvement tasks board for the ongoing release cycle.
-     The following steps can be taken:
-      - Edit improvement tasks board name for current cycle (e.g. `CAPI vX.Y release improvement tasks`)
-      - Add/move all individual missing issues to the board
```
Comment (chrischdi) [inline docs/release/release-tasks.md:296]: I think the replacement for all of these are missing?

**Comment by sbueringer** [comment]
@dhij do you have time to address the finding?

**Comment by dhij** [comment]
@sbueringer thank you for the reminder, i will have a look at it now 👀

File: docs/release/release-tasks.md (hunk 2 above)
Comment (dhij) [reply docs/release/release-tasks.md:296]: @chrischdi Thank you for pointing that out, I missed those points. I have added them. I will add a summary of changes below 👍

**Comment by dhij** [comment]
@chrischdi @sbueringer 👋 I have added the missing information and fixed the existing relative paths to the `release-tasks.md` file that does not exist anymore.

I also have a follow-up PR https://github.com/kubernetes-sigs/cluster-api/pull/10858 as Chris suggested to fix the other existing paths to `release-tasks.md`. Those changes could not be included in this same PR since they are absolute paths that will not exist until this PR is merged. Once this PR is merged, I will do a rebase on that PR 👍

File: docs/release/role-handbooks/release-lead/README.md [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+## Overview
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Responsibilities](#responsibilities)
+- [Tasks](#tasks)
+  - [Finalize release schedule and team](#finalize-release-schedule-and-team)
+  - [Add/remove release team members](#addremove-release-team-members)
+  - [Prepare main branch for development of the new release](#prepare-main-branch-for-development-of-the-new-release)
```
Comment (chrischdi) [inline docs/release/role-handbooks/release-lead/README.md:12]: Doesn't this TOC miss items which are there? e.g.
`Optional] Track Release and Improvement tasks`

File: docs/release/release-team.md [hunk 4]
```diff
@@ -67,11 +67,17 @@ As noted above, making changes to  the CAPI release cadence is out of scope for
...
-- **CI Signal/Bug Triage/Automation Manager**: Assumes the responsibility of the quality gate for the release and makes sure blocking issues and bugs are triaged and dealt with in a timely fashion. Helps improve release automation and tools.
-- **Team member**: Any Release Team lead or manager may select one or more additional members to help with their tasks. These team members will help fulfill future Release Team staffing requirements and continue to grow the CAPI community in general.
-*Note*: This is also documented in [Release tasks](./release-tasks.md) together with a mapping to specific tasks.
+**Notes**:
+
+* The examples in these documents are based on the v1.6 release cycle.
+* The documents focus on tasks that are done for every release. One-time improvement tasks are out of scope.
+* If a task is prefixed with `[Track]` it means it should be ensured that this task is done, but the folks with the corresponding role are not responsible to do it themselves.
```
Comment (chrischdi) [inline docs/release/release-team.md:74]: Should we have this note in each handbook?

File: docs/release/release-team.md [hunk 5]
```diff
@@ -67,11 +67,17 @@ As noted above, making changes to  the CAPI release cadence is out of scope for
...
-- **Communications/Docs/Release Notes Manager**: Responsible for communicating key dates to the community, improving release process documentation, and polishing release notes. Also responsible for ensuring the user-facing Netlify book and provider upgrade documentation are up to date.
-- **CI Signal/Bug Triage/Automation Manager**: Assumes the responsibility of the quality gate for the release and makes sure blocking issues and bugs are triaged and dealt with in a timely fashion. Helps improve release automation and tools.
-- **Team member**: Any Release Team lead or manager may select one or more additional members to help with their tasks. These team members will help fulfill future Release Team staffing requirements and continue to grow the CAPI community in general.
-*Note*: This is also documented in [Release tasks](./release-tasks.md) together with a mapping to specific tasks.
+**Notes**:
+
+* The examples in these documents are based on the v1.6 release cycle.
+* The documents focus on tasks that are done for every release. One-time improvement tasks are out of scope.
```
Comment (chrischdi) [inline docs/release/release-team.md:73]: Does this still make sense as is?

**Comment by sbueringer** [comment]
@dhij Do you have time to address the findings above?

**Comment by dhij** [comment]
@sbueringer 👋 sorry for the delay, yes i will get to it 🙂

File: docs/release/role-handbooks/release-lead/README.md (hunk 3 above)
Comment (dhij) [reply docs/release/role-handbooks/release-lead/README.md:14]: Thank you for pointing that out, I addressed this in my most recent commits.

File: docs/release/release-team.md (hunk 4 above)
Comment (dhij) [reply docs/release/release-team.md]: I moved this bullet point to each of the handbooks in my most recent commits 👍

File: docs/release/release-team.md (hunk 5 above)
Comment (dhij) [reply docs/release/release-team.md]: I removed this line as it is not relevant anymore.

**Comment by dhij** [comment]
@chrischdi @sbueringer 👋 i addressed your comments and would appreciate another review 🙏 

There were [3 new commits to the release-task.md](https://github.com/kubernetes-sigs/cluster-api/commits/main/docs/release/release-tasks.md) in the meantime and since the `release-task.md` file is removed as part of this PR, I had to manually cherry-pick the changes.

**Comment by chrischdi** [comment]
/lgtm
/approve

Will do a small PR on-top after merging.

**Pull request merged**